rust/Makefile

```diff
@@ -60,11 +61,73 @@ alloc-cfgs = \
     --cfg no_sync \
     --cfg no_thin
 
+proc_macro2-skip_flags := \
+    --edition=2021 \
+    -Drust_2018_idioms \
```

Comment:
Maybe do `--cap-lints allow` for all third-party crates? This overrides any `-D` arguments. It is also what cargo passes for all non-local crates.